packages/firestore/src/local/remote_document_change_buffer.ts

```diff
@@ -67,8 +68,14 @@ export abstract class RemoteDocumentChangeBuffer {
 
   /** Buffers a `RemoteDocumentCache.addEntry()` call. */
   addEntry(maybeDocument: MaybeDocument): void {
-    const changes = this.assertChanges();
-    this.changes = changes.insert(maybeDocument.key, maybeDocument);
+    this.assertChanges();
+    this.changes!.set(maybeDocument.key, maybeDocument);
+  }
+
+  /** Buffers a `RemoteDocumentCache.removeEntry()` call. */
```

Comment:
Is the contract here that you must call getEntry() prior to calling removeEntry() [in order for `documentSizes` to be populated so we can update the size delta]?  Can you document that here (and on addEntry() which presumably has the same contract)?